.husky/pre-commit

```diff
@@ -2,3 +2,8 @@
 . "$(dirname "$0")/_/husky.sh"
 
 npx lint-staged
+
+# update the API report
+mkdir -p ./etc
+npm run build-types -- -l
+git add ./etc/oidc-client-ts.api.md
```

Comment:
I have the feeling that `git add ./docs/oidc-client-ts.api.md` will lead to merge conflicts or parts in a merge request we do not want to merge from "yet unknown" developers, then that developer needs to do a rebase, which not everybody can do, .... It might be better when we do this manually (e.g. when the CI fails) and particularly before we do a release (to refresh docs).

I am not sure if we want to have husky at all. First i was worried it wont work under Windows, but it looks like it does. But it duplicates stuff which the CI is already handling for us. Thus it makes stuff more complicated. husky might be good for private repos where all involved developers know about it, but this is an open repo and we will have merge request from outside...

Lets remove husky for now unless there is a real use case! Lets keep stuff simple and stupid.

Reply:
Sure, but I at least think lint-staged is still useful at any scale, it's simply better dev ergonomics. CI isn't fixing and committing code that was edited, only your local machine can do that. For example #74 had some back and forth that could have been avoided. The CI step is definitely still needed, though, because merging and rebasing can give rise to lint errors that don't run through the pre-commit hook.

Reply:
ok, but lets keep it only for lint-staged...